core/application/tests/bloodDonationWorkflow/BloodDonationService.test.ts

```diff
@@ -0,0 +1,133 @@
+import { BloodDonationService } from '../../bloodDonationWorkflow/BloodDonationService'
+import { DonationDTO } from '../../../../commons/dto/DonationDTO'
+import Repository from '../../technicalImpl/policies/repositories/Repository'
+import { generateUniqueID } from '../../utils/idGenerator'
+import { generateGeohash } from '../../utils/geohash'
+import { validateInputWithRules } from '../../utils/validator'
+import BloodDonationOperationError from '../../bloodDonationWorkflow/BloodDonationOperationError'
+import { donationAttributes } from '../mocks/mockDonationRequestData'
+import { mockRepository as importedMockRepository } from '../mocks/mockRepositories'
+
+jest.mock('../../utils/idGenerator', () => ({
+  generateUniqueID: jest.fn()
+}))
+
+jest.mock('../../utils/geohash', () => ({
+  generateGeohash: jest.fn()
+}))
+
+jest.mock('../../utils/validator', () => ({
+  validateInputWithRules: jest.fn()
+}))
+
+describe('BloodDonationService', () => {
+  const bloodDonationService = new BloodDonationService()
+  const bloodDonationRepository = importedMockRepository as jest.Mocked<Repository<DonationDTO>>
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('createBloodDonation', () => {
+    test('should create a blood donation and return success message', async() => {
+      (generateUniqueID as jest.Mock).mockReturnValue('uniqueID');
+      (generateGeohash as jest.Mock).mockReturnValue('geohash123');
+      (validateInputWithRules as jest.Mock).mockReturnValue(null)
+
+      const result = await bloodDonationService.createBloodDonation(donationAttributes, bloodDonationRepository)
+
+      expect(generateUniqueID).toHaveBeenCalled()
+      expect(generateGeohash).toHaveBeenCalledWith(donationAttributes.latitude, donationAttributes.longitude)
+      expect(bloodDonationRepository.create).toHaveBeenCalledWith(expect.objectContaining({
+        id: 'uniqueID',
+        status: 'accepted',
```

Comment:
use enum

Reply:
Create an enum and use it in code. [Commit](https://github.com/CraftsmenLtd/BloodConnect/pull/177/commits/be92e1051e74df3cdc6edffb1aad6cfb725c40e8)